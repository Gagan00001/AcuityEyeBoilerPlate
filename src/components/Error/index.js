import React from 'react';
import PropTypes from 'prop-types';
import isArray from 'lodash/isArray';

function Error(props) {
  const { messages } = props;
  return (
    <div>
      {isArray(messages) ? (
        messages.map((item, index) => <div key={`error${index}`}>{item}</div>)
      ) : (
        <span>{messages}</span>
      )}
    </div>
  );
}

Error.defaultProps = {
  messages: '',
};

Error.propTypes = {
  messages: PropTypes.string,
};

export default Error;
