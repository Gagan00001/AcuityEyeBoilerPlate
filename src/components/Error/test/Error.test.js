import React from 'react';
import { render } from '@testing-library/react';
import { unmountComponentAtNode } from 'react-dom';
import Error from '..';

let container = null;

beforeEach(() => {
  // setup a DOM element as a render target
  container = document.createElement('div');
  document.body.appendChild(container);
});

afterEach(() => {
  // cleanup on exiting
  unmountComponentAtNode(container);
  container.remove();
  container = null;
});

const messages = ['Hi this is error.', 'How are you?'];
const RenderErrorComponent = (props) => (<Error {...props} />);

describe('Error component testing', () => {
  test('should match snapshot with message.', () => {
    const { container: errorContainer } = render(
      <RenderErrorComponent messages="Hii I am error" />, container,
    );
    expect(errorContainer).toMatchSnapshot();
  });
  test('should match snapshot with multiple messages.', () => {
    const { container: errorContainer } = render(
      <RenderErrorComponent messages={messages} />, container,
    );
    expect(errorContainer).toMatchSnapshot();
  });
  test('should match snapshot with no messages.', () => {
    const { container: errorContainer } = render(
      <RenderErrorComponent />, container,
    );
    expect(errorContainer).toMatchSnapshot();
  });
});
