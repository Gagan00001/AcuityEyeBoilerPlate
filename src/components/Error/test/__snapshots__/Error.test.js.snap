// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Error component testing should match snapshot with message. 1`] = `
<div>
  <div>
    <span>
      Hii I am error
    </span>
  </div>
</div>
`;

exports[`Error component testing should match snapshot with multiple messages. 1`] = `
<div>
  <div>
    <div>
      Hi this is error.
    </div>
    <div>
      How are you?
    </div>
  </div>
</div>
`;

exports[`Error component testing should match snapshot with no messages. 1`] = `
<div>
  <div>
    <span />
  </div>
</div>
`;
